### PR TITLE
199/full text search

### PR DESCRIPTION
**Pull request by FabioBurillo**
Issues #505 and #506 were implemented in this pr.

List of features:
Search results are shown in tabs by type. 
Tabs are shown only if there is a result for this type.
Each tab has its own fetching list and is loaded when clicked.
Each fetching list has its own query depending on type.
Updated text translations and search result count.
Loading icon while all-tab search results are loaded.
Each fetchinglist-item has icon displaying its type.
Items route to correct page on click.

Not only related to search results:
Removed flex options on global .content-css-class to give a smoother experience (content is no longer moving around when page is loaded)
All icons were refactored to be defined under a single enum in entity.interface.ts file.

### CODE REVIEW

File: app/src/app/app.component.scss [hunk 1]
```diff
@@ -14,10 +14,6 @@
 
 .content {
   position: relative;
-  flex: 1;
```
Comment (EeveesEyes) [inline app/src/app/app.component.scss:17]: @tpachmann @DPhuongHda can you please check this? I'm not sure whether this breaks other parts of the application. The intention was to remove the empty space on top of 'non-scrollable' components
Before:
non-scrollable | scrollable
:----:|:----:|
image | image

Comment (DPhuongHda) [reply app/src/app/app.component.scss:17]: As far as I can see this change didn't seem to affect other parts of the website

File: app/src/app/pages/search-result/search-result.component.html [hunk 2]
```diff
@@ -1,45 +1,83 @@
...
+  <div *ngIf='allTabLoaded'>
+    <ul class='nav nav-tabs' *ngIf='entityTabs.length != 0'>
+      <ng-container *ngFor='let tab of entityTabs'>
+        <li class='nav-item' *ngIf='tab.loaded'>
+          <a class='nav-link' [ngClass]='{ active: tab.active }' data-toggle='tab' href='#{{ tab.type }}_tab'
+             (click)='tabClicked(tab)'>
+            <i class='mr-1 fa fa-{{ tab.icon }}'></i>
+            <ng-container i18n='Entity Type Labels@@entity_labels' *ngIf='tab.type; let type'>
+              {type, select, all {All} artwork {Artwork} motif {Motif} artist {Artist} location {Location} genre {Genre} movement {Movement} material
+                {Material} main_motif {Main Motif} }
```
Comment (EeveesEyes) [inline app/src/app/pages/search-result/search-result.component.html:60]: check translation string für class

File: app/src/app/pages/search-result/search-result.component.ts [hunk 3]
```diff
@@ -1,84 +1,113 @@
...
-        this.searchResults.push(await this.getSearchResults(params.motif, EntityType.MOTIF, EntityIcon.MOTIF));
+        this.searchObjects.push(await this.getSearchObjects(params.motif, EntityType.MOTIF, EntityIcon.MOTIF));
       }
       if (params.location) {
-        this.searchResults.push(await this.getSearchResults(params.location, EntityType.LOCATION, EntityIcon.LOCATION));
+        this.searchObjects.push(await this.getSearchObjects(params.location, EntityType.LOCATION, EntityIcon.LOCATION));
       }
       if (params.material) {
-        this.searchResults.push(await this.getSearchResults(params.material, EntityType.MATERIAL, EntityIcon.MATERIAL));
+        this.searchObjects.push(await this.getSearchObjects(params.material, EntityType.MATERIAL, EntityIcon.MATERIAL));
```
Comment (EeveesEyes) [inline app/src/app/pages/search-result/search-result.component.ts:92]: Please parallelize these calls by collecting all promises and resolve all together by calling Promise.all()

**Review by EeveesEyes — CHANGES_REQUESTED** [review]
This looks promising, together with the discussed 'issues'
- image not loading in all tab in some cases.
- change icon for artworks on artists and movements page
- icons on tabs of search result page

and my requested changes, this should be ready to merge :) nice job

File: app/src/app/pages/search-result/search-result.component.ts (hunk 3 above)
Comment (vinoroidski) [reply app/src/app/pages/search-result/search-result.component.ts:92]: @EeveesEyes : we were unable to get out code working here and have to start with the release preparation. We have to look at this topic in our next release as part of larger refactoring.

**Review by EeveesEyes — APPROVED** [review]
looks good :+1:

**Pull request merged**